ProductDetails.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from 'react';
+import { View, Text, StyleSheet, TouchableOpacity, Image, SafeAreaView, Platform, StatusBar, ScrollView } from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { Ionicons } from '@expo/vector-icons';
+import { useFetch } from './useFetch';
+
+const ProductDetails = ({ route, navigation }) => {
+  const { productId } = route.params;
+  const [isFavorite, setIsFavorite] = useState(false);
+  const { data: productData, loading } = useFetch(`https://rn-products-1d8a6-default-rtdb.firebaseio.com/products/${productId}.json`);
+
+  useEffect(() => {
+    // Check if the product is already favorited in async storage
+    checkFavoriteStatus();
+  }, []);
+
+  const checkFavoriteStatus = async () => {
+    try {
+      const favorites = await AsyncStorage.getItem('favorites');
+      setIsFavorite(favorites.includes(productId));
+    } catch (error) {
+      console.error("Error checking favorite status: ", error);
+    }
+  };
+
+  const toggleFavorite = async () => {
+    try {
+      const favorites = await AsyncStorage.getItem('favorites');
+      let favoriteProducts = JSON.parse(favorites) || [];
+
+      if (isFavorite) {
+        favoriteProducts = favoriteProducts.filter(id => id !== productId);
+      } else {
+        favoriteProducts.push(productId);
+      }
+      await AsyncStorage.setItem('favorites', JSON.stringify(favoriteProducts));
+      setIsFavorite(!isFavorite);
+
+      await checkFavoriteStatus();
+    } catch (error) {
+      console.error("Error toggling favorite: ", error);
+    }
+  };
+
+  if (loading) {
+    return <Text>Loading...</Text>;
+  }
+
+  if (!productData) {
+    return <Text>No product data available</Text>;
+  }
+
+  return (
+    <View style={styles.container}>
+      <StatusBar barStyle="dark-content" />
+      <TouchableOpacity style={styles.backButton} onPress={() => navigation.goBack()}>
+        <Text>Back</Text>
+      </TouchableOpacity>
+      <TouchableOpacity
+        style={styles.favoriteButton}
```

Comment:
If you put the favorite icon in the header itself, react navigation would make sure it's always visible.